app/pages/register/settlors/living_settlor/business/SettlorBusinessAddressInternationalPage.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pages.register.settlors.living_settlor.business
+
+import models.core.pages.InternationalAddress
+import pages.QuestionPage
+import play.api.libs.json.JsPath
+import sections.LivingSettlors
+
+final case class SettlorBusinessAddressInternationalPage(index : Int) extends QuestionPage[InternationalAddress] {
+
+  override def path: JsPath = LivingSettlors.path \ index \ toString
+
+  override def toString: String = "address"
```

Comment:
Using the same key for ukAddress and internationalAddress will cause issues if the user goes through a Change journey, as the cleanup logic in SettlorBusinessAddressUKYesNoPage will remove both.

Reply:
Agreed, but this has been done for all existing settlors and requires a big rework of how the mapper reads addresses for everyone so I'd suggest this is something that needs to go with the change journey ticket when that is being worked on as there is no change journey for any of them at the moment.

Reply:
Actually this would happen in a normal journey too right? If you select yes to "UK address yes no", then enter a UK address, then go back twice and select yes again it would remove the address.

The rework is actually quite small, just requires a read on 'ukAddress' orElse a read on 'internationalAddress'. See https://github.com/hmrc/trusts-frontend/blob/9df60ed0524a07b034a7ceca7ecef39d3e1706de/app/mapping/reads/PropertyOrLandAsset.scala#L55

Reply:
Resolved now